### PR TITLE
[AI-FSSDK] [FSSDK-12337] Add Feature Rollout support

### PR DESCRIPTION
**Pull request by FarhanAnjum-opti**
## Summary

Adds Feature Rollout support to the Python SDK. Feature Rollouts are a new experiment rule type that combines Targeted Delivery simplicity with A/B test measurement capabilities. During project config parsing, the "everyone else" variation from the flag's rollout is injected into any experiment with type "feature_rollout", enabling correct evaluation without changes to decision logic.

## Changes

- Added optional `type` string field to the Experiment data model
- Added config parsing logic to inject the "everyone else" rollout variation into feature_rollout experiments
- Added traffic allocation entry (endOfRange=10000) for the injected variation
- Added `_get_everyone_else_variation` helper method to extract the last rollout rule's first variation
- Added 16 unit tests covering feature rollout injection, edge cases, and backward compatibility

## Jira Ticket

[FSSDK-12337](https://optimizely-ext.atlassian.net/browse/FSSDK-12337)

[FSSDK-12337]: https://optimizely-ext.atlassian.net/browse/FSSDK-12337?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: optimizely/project_config.py [hunk 1]
```diff
@@ -232,6 +232,34 @@ def __init__(self, datafile: str | bytes, logger: Logger, error_handler: Any):
                 self.experiment_feature_map[exp_id] = [feature.id]
                 rules.append(self.experiment_id_map[exp_id])
 
+            # Feature Rollout support: inject the "everyone else" variation
+            # into any experiment with type == "feature_rollout"
+            everyone_else_variation = self._get_everyone_else_variation(feature)
+            if everyone_else_variation is not None:
```
Comment (jaeopt) [inline optimizely/project_config.py:255]: We need this condition to the ticket prompt -
if everyone_else is null, we need handle specially. Let's discuss.

File: optimizely/project_config.py [hunk 2]
```diff
@@ -232,6 +232,34 @@ def __init__(self, datafile: str | bytes, logger: Logger, error_handler: Any):
...
+                        })
+                        var_entity = entities.Variation(
+                            id=everyone_else_variation['id'],
+                            key=everyone_else_variation['key'],
+                            featureEnabled=bool(everyone_else_variation.get('featureEnabled', False)),
+                            variables=cast(
+                                Optional[list[entities.Variable]],
+                                everyone_else_variation.get('variables'),
+                            ),
+                        )
```
Comment (jaeopt) [inline optimizely/project_config.py]: Can this be returned by _get_everyone_else_variation() instead of building redeatedly?
We can try add this to the prompt.

File: optimizely/project_config.py [hunk 3]
```diff
@@ -667,6 +698,41 @@ def get_rollout_from_id(self, rollout_id: str) -> Optional[entities.Layer]:
...
+        rollout = self.get_rollout_from_id(flag.rolloutId)
+        if not rollout or not rollout.experiments:
+            return None
+
+        everyone_else_rule = rollout.experiments[-1]
+        variations = everyone_else_rule.get('variations', [])
+        if not variations:
+            return None
+
+        variation_dict = variations[0]
```
Comment (muzahidul-opti) [inline optimizely/project_config.py:742]: Is it save to access? What happen if it is empty

File: optimizely/helpers/types.py [hunk 4]
```diff
@@ -117,6 +117,8 @@ class CmabDict(BaseEntity):
     trafficAllocation: int
 
 
+ExperimentType = Literal['a/b', 'mab', 'cmab', 'feature_rollout']
```
Comment (jaeopt) [inline optimizely/helpers/types.py]: flags have these types -
class RuleTypes(enum.Enum):
    targeted_delivery = "targeted_delivery"
    ab = "a/b"
    multi_armed_bandit = "multi_armed_bandit"
    contextual_multi_armed_bandit = "contextual_multi_armed_bandit"

if we want to keep short names - let's consistent. what about 
{"ab", "mab", "cmab", "td", "fr"}

Comment (jaeopt) [reply optimizely/helpers/types.py]: We can specify these types in the ticket.
Also need to update TDD as well.

File: tests/test_config.py [hunk 5]
```diff
@@ -1545,3 +1545,415 @@ def test_holdout_initialization__only_processes_running_holdouts(self):
...
+        opt = optimizely.Optimizely(json.dumps(datafile))
+        config = opt.config_manager.get_config()
+
+        experiment = config.experiment_id_map['exp_fr']
+
+        # Without a rollout, no injection should occur
+        self.assertEqual(len(experiment.variations), 1)
+        self.assertEqual(len(experiment.trafficAllocation), 1)
+
+    def test_feature_rollout_everyone_else_is_last_rollout_rule(self):
```
Comment (jaeopt) [inline tests/test_config.py]: what is this test about? it looks redundant?

File: optimizely/project_config.py (hunk 3 above)
Comment (FarhanAnjum-opti) [reply optimizely/project_config.py:742]: It is safe.  Line 722-723 ensures its not empty and if empty None is returned.

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks good. A couple of changes suggested.

**Review by esrakartalOpt — APPROVED** [review]
All comments are implemented. Looks good to me 👍

**Review by jaeopt — APPROVED** [review]
LGTM
